### PR TITLE
Add validation annotations and a check

### PR DESCRIPTION
**Pull request by keks**

### CODE REVIEW

**Review by franziskuskiefer — APPROVED** [review]
Thanks, lgtm

File: openmls/src/group/public_group/validation.rs [hunk 1]
```diff
@@ -491,6 +492,7 @@ impl PublicGroup {
         &self,
         proposal_queue: &ProposalQueue,
     ) -> Result<(), ExternalCommitValidationError> {
+        // [valno0401](https://validation.openmls.tech/#valno0401)
```
Comment (kkohbrok) [inline openmls/src/group/public_group/validation.rs:495]:
```suggestion
        // [valno0401](https://validation.openmls.tech/#valn0401)
```

File: openmls/src/group/public_group/validation.rs [hunk 2]
```diff
@@ -503,6 +505,7 @@ impl PublicGroup {
         }
 
         // ValSem242: External Commit must only cover inline proposal in allowlist (ExternalInit, Remove, PreSharedKey)
+        // [valno0404](https://validation.openmls.tech/#valno0404)
```
Comment (kkohbrok) [inline openmls/src/group/public_group/validation.rs:508]:
```suggestion
        // [valno0404](https://validation.openmls.tech/#valn0404)
```

File: openmls/src/group/public_group/staged_commit.rs [hunk 3]
```diff
@@ -171,6 +173,24 @@ impl PublicGroup {
         Ok((commit, proposal_queue, sender_index))
     }
 
+    // Check that no external init proposal occurs. Needed only for regular commits.
+    // [valno0310](https://validation.openmls.tech/#valn0310)
```
Comment (kkohbrok) [inline openmls/src/group/public_group/staged_commit.rs:177]:
```suggestion
    // [valn0310](https://validation.openmls.tech/#valn0310)
```

File: openmls/src/group/mls_group/staged_commit.rs [hunk 4]
```diff
@@ -72,6 +72,7 @@ impl MlsGroup {
         };
 
         // Prepare the PskSecret
+        // Fails if PSKs are missing ([valno1205](https://validation.openmls.tech/#valn1205))
```
Comment (kkohbrok) [inline openmls/src/group/mls_group/staged_commit.rs:75]:
```suggestion
        // Fails if PSKs are missing ([valn1205](https://validation.openmls.tech/#valn1205))
```

**Review by kkohbrok — CHANGES_REQUESTED** [review]
Just a few nits, otherwise looks good!

**Comment by keks** [comment]
Oh, good catch! Yeah I forgot we changed the naming scheme at some point. The new one is much better :)

**Review by kkohbrok — APPROVED** [review]
Thanks!

**Pull request merged**